guides/search.md

```diff
@@ -0,0 +1,345 @@
+# Search
+
+OpenSearch provides a powerful search API that allows you to search for documents in an index. The search API supports a number of parameters that allow you to customize the search operation. In this guide, we will explore the search API and its parameters.
+
+# Setup
+
+Let's start by creating an index and adding some documents to it:
+
+```rust
+let url = Url::parse("https://localhost:9200")?;
+let credentials = Credentials::Basic("admin".into(), "admin".into());
+let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url))
+    .cert_validation(CertificateValidation::None)
+    .auth(credentials)
+    .build()?;
+let client = OpenSearch::new(transport);
+client
+  .indices()
+  .create(IndicesCreateParts::Index("movies"))
+  .send()
+  .await?;
+
+for n in 0..11 {
+    client
+        .index(IndexParts::IndexId("movies", i))
+        .body(json!({
+            "id": i,
```

Comment:
You do not need to specify `"id": i` here as it's already set as part of the IndexParts above